File: scripts/validate-posts.js
import { listPostDirectories, readPost, validatePost } from "./lib.js";

const directories = listPostDirectories();
if (directories.length === 0) {
  console.log("검사할 게시글이 없습니다.");
  process.exit(0);
}

const errors = [];
for (const dir of directories) {
  try {
    const post = readPost(dir);
    errors.push(...validatePost(post));
  } catch (error) {
    errors.push(error.message);
  }
}

if (errors.length > 0) {
  console.error("\n게시글 검증 실패:");
  for (const error of errors) console.error(`- ${error}`);
  process.exit(1);
}

console.log(`${directories.length}개 게시글 검증 완료.`);

File: living-fix-note-python-v2/scripts/validate_posts.py
from common import post_directories, read_post, validate_post


def main() -> None:
    directories = post_directories()
    if not directories:
        print("검사할 게시글이 없습니다.")
        return

    errors: list[str] = []
    for directory in directories:
        try:
            metadata, content = read_post(directory)
            errors.extend(validate_post(directory, metadata, content))
        except ValueError as exc:
            errors.append(str(exc))

    if errors:
        print("게시글 검증 실패:")
        for error in errors:
            print(f"- {error}")
        raise SystemExit(1)

    print(f"{len(directories)}개 게시글 검증 완료.")


if __name__ == "__main__":
    main()
